2025-01-04 01:09:21,290 : INFO : root : Loading config.
2025-01-04 01:09:21,295 : INFO : YamlMultifileConfigLoader : Machine config file #1: config.yaml
2025-01-04 01:09:21,295 : INFO : ConfigProcessor : Loading config from cache: C:\Users\marcu\AppData\Local\Temp\62ffdcdfb903f5d0df8285e2704fd504.mpf_cache
2025-01-04 01:09:21,379 : WARNING : ConfigProcessor : Config file in cache changed: C:\Users\marcu\Documents\RA2-Pinball\modes\attract\config\attract.yaml
2025-01-04 01:09:21,379 : INFO : ConfigProcessor : Loading config from file C:\Users\marcu\mpf-env\Lib\site-packages\mpf\modes\attract\config\attract.yaml.
2025-01-04 01:09:21,381 : INFO : ConfigProcessor : Loading config: C:\Users\marcu\mpf-env\Lib\site-packages\mpf\modes\attract\config\attract.yaml
2025-01-04 01:09:21,381 : INFO : ConfigProcessor : Loading config from file C:\Users\marcu\Documents\RA2-Pinball\modes\attract\config\attract.yaml.
2025-01-04 01:09:21,392 : INFO : ConfigProcessor : Loading config: C:\Users\marcu\Documents\RA2-Pinball\modes\attract\config\attract.yaml
2025-01-04 01:09:21,393 : INFO : ConfigProcessor : Config file cache created: C:\Users\marcu\AppData\Local\Temp\aa5bf5e7d7698d57ef5cf546c9fe3902.mpf_cache
2025-01-04 01:09:21,393 : INFO : ConfigProcessor : Loading config from cache: C:\Users\marcu\AppData\Local\Temp\8585e1b9e1c9a830f5d1d7fad5239a38.mpf_cache
2025-01-04 01:09:21,394 : INFO : ConfigProcessor : Loading config from cache: C:\Users\marcu\AppData\Local\Temp\aa5bf5e7d7698d57ef5cf546c9fe3902.mpf_cache
2025-01-04 01:09:21,403 : INFO : ConfigProcessor : Loading config from cache: C:\Users\marcu\AppData\Local\Temp\a5167e1ab4e20de4bfe590f9baa63622.mpf_cache
2025-01-04 01:09:21,404 : INFO : Machine : Mission Pinball Framework Core Engine v0.80.0.dev6
2025-01-04 01:09:21,404 : INFO : Machine : Command line arguments: {'no_load_cache': False, 'create_config_cache': True, 'bcp': True, 'configfile': ['config.yaml'], 'force_assets_load': False, 'platform_integration_test': False, 'jsonlogging': False, 'logfile': 'logs\\2025-01-04-01-09-21-mpf-DESKTOP-P33ACKK.log', 'pause': False, 'production': False, 'text_ui': False, 'loglevel': 15, 'consoleloglevel': 20, 'force_platform': 'virtual', 'syslog_address': None, 'full_logfile_path': 'C:\\Users\\marcu\\Documents\\RA2-Pinball\\logs\\2025-01-04-01-09-21-mpf-DESKTOP-P33ACKK.log'}
2025-01-04 01:09:21,404 : INFO : Machine : MPF path: C:\Users\marcu\mpf-env\Lib\site-packages\mpf
2025-01-04 01:09:21,405 : INFO : Machine : Machine path: C:\Users\marcu\Documents\RA2-Pinball
2025-01-04 01:09:21,405 : INFO : Machine : Platform: win32
2025-01-04 01:09:21,405 : INFO : Machine : Python executable location: C:\Users\marcu\mpf-env\Scripts\python.exe
2025-01-04 01:09:21,405 : INFO : Machine : Python version: 3.12.7 (64-bit)
2025-01-04 01:09:21,407 : INFO : Machine : Initializing MPF...
2025-01-04 01:09:21,483 : INFO : EventManager : Event: ======'machine_var_credits_string'====== Args={'value': 'FREE PLAY', 'prev_value': None, 'change': True}
2025-01-04 01:09:21,502 : INFO : EventManager : Event: ======'machine_var_master_volume'====== Args={'value': 0.5, 'prev_value': None, 'change': True}
2025-01-04 01:09:21,502 : INFO : EventManager : Event: ======'machine_var_mpf_version'====== Args={'value': 'Mission Pinball Framework v0.80.0.dev6', 'prev_value': None, 'change': True}
2025-01-04 01:09:21,502 : INFO : EventManager : Event: ======'machine_var_mpf_extended_version'====== Args={'value': 'Mission Pinball Framework v0.80.0.dev6, Config version:6, Show version: 6, BCP version:1.1', 'prev_value': None, 'change': True}
2025-01-04 01:09:21,503 : INFO : EventManager : Event: ======'machine_var_python_version'====== Args={'value': '3.12.7', 'prev_value': None, 'change': True}
2025-01-04 01:09:21,611 : INFO : EventManager : Event: ======'machine_var_platform'====== Args={'value': 'Windows-10-10.0.19045-SP0', 'prev_value': None, 'change': True}
2025-01-04 01:09:21,611 : INFO : EventManager : Event: ======'machine_var_platform_system'====== Args={'value': 'Windows', 'prev_value': None, 'change': True}
2025-01-04 01:09:21,611 : INFO : EventManager : Event: ======'machine_var_platform_release'====== Args={'value': '10', 'prev_value': None, 'change': True}
2025-01-04 01:09:21,611 : INFO : EventManager : Event: ======'machine_var_platform_version'====== Args={'value': '10.0.19045', 'prev_value': None, 'change': True}
2025-01-04 01:09:21,611 : INFO : EventManager : Event: ======'machine_var_platform_machine'====== Args={'value': 'AMD64', 'prev_value': None, 'change': True}
2025-01-04 01:09:21,611 : INFO : EventManager : Event: ======'machine_var_log_file_path'====== Args={'value': 'C:\\Users\\marcu\\Documents\\RA2-Pinball\\logs\\2025-01-04-01-09-21-mpf-DESKTOP-P33ACKK.log', 'prev_value': None, 'change': True}
2025-01-04 01:09:21,611 : INFO : EventManager : Event: ======'init_phase_1'====== Args={}
2025-01-04 01:09:21,626 : INFO : EventManager : Event: ======'init_phase_2'====== Args={}
2025-01-04 01:09:21,627 : ERROR : asyncio : Exception in callback EventManager._queue_task_done(<Task finishe...sounds None')>)
handle: <Handle EventManager._queue_task_done(<Task finishe...sounds None')>)>
Traceback (most recent call last):
  File "C:\Users\marcu\AppData\Local\Programs\Python\Python312\Lib\asyncio\events.py", line 88, in _run
    self._context.run(self._callback, *self._args)
  File "C:\Users\marcu\mpf-env\Lib\site-packages\mpf\core\events.py", line 795, in _queue_task_done
    future.result()
  File "C:\Users\marcu\mpf-env\Lib\site-packages\mpf\core\events.py", line 708, in _run_handlers_sequential
    handler.callback(queue=queue, **merged_kwargs)
  File "C:\Users\marcu\mpf-env\Lib\site-packages\mpf\core\mode_controller.py", line 97, in initialize_modes
    item.method(config=mode.config[item.config_section],
  File "C:\Users\marcu\mpf-env\Lib\site-packages\mpf\core\config_player.py", line 150, in process_mode_config
    config = self.validate_config(config)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\marcu\mpf-env\Lib\site-packages\mpf\core\config_player.py", line 113, in validate_config
    validated_config[event] = self.validate_config_entry(settings, event)
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\marcu\mpf-env\Lib\site-packages\mpf\config_players\device_config_player.py", line 44, in validate_config_entry
    raise AssertionError(
AssertionError: Invalid settings for player mode_attract_started:sounds None
2025-01-04 01:09:21,698 : INFO : Machine : Starting the main run loop with active modes: []
2025-01-04 01:09:21,698 : WARNING : Machine : Attract mode is not active, game will not be playable. Please check your attract mode configuration.
2025-01-04 01:09:21,699 : INFO : Machine : Shutting down...
2025-01-04 01:09:21,699 : INFO : EventManager : Event: ======'shutdown'====== Args={}
2025-01-04 01:09:22,708 : ERROR : Machine : Runtime Exception
Traceback (most recent call last):
  File "C:\Users\marcu\mpf-env\Lib\site-packages\mpf\core\machine.py", line 808, in _run_loop
    raise self._exception['exception']
  File "C:\Users\marcu\AppData\Local\Programs\Python\Python312\Lib\asyncio\events.py", line 88, in _run
    self._context.run(self._callback, *self._args)
  File "C:\Users\marcu\mpf-env\Lib\site-packages\mpf\core\events.py", line 795, in _queue_task_done
    future.result()
  File "C:\Users\marcu\mpf-env\Lib\site-packages\mpf\core\events.py", line 708, in _run_handlers_sequential
    handler.callback(queue=queue, **merged_kwargs)
  File "C:\Users\marcu\mpf-env\Lib\site-packages\mpf\core\mode_controller.py", line 97, in initialize_modes
    item.method(config=mode.config[item.config_section],
  File "C:\Users\marcu\mpf-env\Lib\site-packages\mpf\core\config_player.py", line 150, in process_mode_config
    config = self.validate_config(config)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\marcu\mpf-env\Lib\site-packages\mpf\core\config_player.py", line 113, in validate_config
    validated_config[event] = self.validate_config_entry(settings, event)
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\marcu\mpf-env\Lib\site-packages\mpf\config_players\device_config_player.py", line 44, in validate_config_entry
    raise AssertionError(
AssertionError: Invalid settings for player mode_attract_started:sounds None
2025-01-04 01:09:22,750 : INFO : root : MPF run loop ended.
